### PR TITLE
Pref migration for translate

### PR DESCRIPTION
**Pull request by atuchin-m**
Resolves https://github.com/brave/brave-browser/issues/23155
Resolves https://github.com/brave/brave-browser/issues/23139

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: chromium_src/components/translate/core/browser/translate_language_list.h [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_CHROMIUM_SRC_COMPONENTS_TRANSLATE_CORE_BROWSER_TRANSLATE_LANGUAGE_LIST_H_
+#define BRAVE_CHROMIUM_SRC_COMPONENTS_TRANSLATE_CORE_BROWSER_TRANSLATE_LANGUAGE_LIST_H_
+
+namespace translate {
+class TranslateLanguageList;
+using BraveTranslateLanguageList = TranslateLanguageList;
```
Comment (goodov) [inline chromium_src/components/translate/core/browser/translate_language_list.h:11]: pls keep class name prefix the same as in chromium. only append to the class name, don't prepend.
`TranslateLanguageList_BraveImpl`

File: chromium_src/components/translate/core/browser/translate_language_list.h [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+#define resource_requests_allowed_ \
+  resource_requests_allowed_;      \
+  friend BraveTranslateLanguageList
+#include "src/components/translate/core/browser/translate_language_list.h"
+#undef resource_requests_allowed_
+#undef TranslateLanguageList
+
+namespace translate {
+
+class TranslateLanguageList : public ChromiumTranslateLanguageList {
```
Comment (goodov) [inline chromium_src/components/translate/core/browser/translate_language_list.h:24]: same. append only.

Comment (atuchin-m) [reply chromium_src/components/translate/core/browser/translate_language_list.h]: Done

File: chromium_src/components/translate/core/browser/translate_language_list.h (hunk 1 above)
Comment (atuchin-m) [reply chromium_src/components/translate/core/browser/translate_language_list.h]: Done

File: browser/translate/brave_translate_prefs_migration.cc [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+#include "brave/components/translate/core/common/brave_translate_features.h"
+#include "components/prefs/pref_registry_simple.h"
+#include "components/prefs/pref_service.h"
+#include "components/translate/core/browser/translate_pref_names.h"
+
+namespace translate {
+
+namespace prefs {
+const char kMigratedToInternalTranslation[] =
+    "translate_migrated_to_internal_translation";
```
Comment (iefremov) [inline browser/translate/brave_translate_prefs_migration.cc:17]: there is a vague convention to prefix our preferences with "brave." :)

Comment (iefremov) [reply browser/translate/brave_translate_prefs_migration.cc:17]: also not super clear what is an internal translation - maybe to "cloud translation"?

**Review by iefremov — APPROVED** [review]

File: browser/translate/brave_translate_prefs_migration.cc (hunk 3 above)
Comment (atuchin-m) [reply browser/translate/brave_translate_prefs_migration.cc]: Renamed to `brave.translate_migrated_from_extension`

**Pull request merged**